core/Symbols.h

```diff
@@ -138,23 +138,25 @@ class Method final {
     bool ignoreInHashing(const GlobalState &gs) const;
     bool isPrintable(const GlobalState &gs) const;
 
+    // Equivalent to `getIntrinsic() != nullptr`, but potentially more efficient.
+    bool hasIntrinsic() const;
+    const IntrinsicMethod *getIntrinsic() const;
+
     ClassOrModuleRef owner;
     NameRef name;
     ClassOrModuleRef rebind;
+    Flags flags;
+    const static uint16_t INVALID_INTRINSIC_OFFSET = 0;
+    const static uint16_t FIRST_VALID_INTRINSIC_OFFSET = 1;
+    uint16_t intrinsicOffset = INVALID_INTRINSIC_OFFSET;
```

Comment:
What do you think about adding a comment here like this assumes we will only ever have < 2^16 intrinsic methods which is reasonable